Code/GraphMol/ChemReactions/SanitizeRxn.cpp

```diff
@@ -324,15 +323,67 @@ void fixReactantTemplateAromaticity(ChemicalReaction &rxn) {
 }
 
 void fixHs(ChemicalReaction &rxn) {
+  {
+    //  if mapped Hydrogens in reactants area mapped to heavy atoms
+    //   keep mappings, in all other cases remove them.
+    //   this allows us to merge query hydrogens atoms 
+
+    std::map<int, bool> mappedToNonHeavyProductAtom;
```

Comment:
of course you're right. I went through that too quickly and thought this was atom indexed.